packages/network-controller/src/create-network-client.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  createAsyncMiddleware,
+  createScaffoldMiddleware,
+  JsonRpcEngine,
+  mergeMiddleware,
+  JsonRpcMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockCacheMiddleware,
+  createBlockRefMiddleware,
+  createBlockRefRewriteMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  createInflightCacheMiddleware,
+  createFetchMiddleware,
+  createRetryOnEmptyMiddleware,
+} from '@metamask/eth-json-rpc-middleware';
+import {
+  providerFromEngine,
+  providerFromMiddleware,
+  SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import type { Hex } from '@metamask/utils/dist';
+import { PollingBlockTracker } from 'eth-block-tracker/dist';
+import { NetworksChainId } from '@metamask/controller-utils';
+
+const SECOND = 1000;
+
+export enum NetworkClientType {
+  Custom = 'custom',
+  Infura = 'infura',
+}
+
+type CustomNetworkConfiguration = {
+  chainId?: Hex;
+  rpcUrl: string;
+  type: NetworkClientType.Custom;
+};
+
+// could improve this type... this is not a very complete list
+export type InfuraNetworkType = 'mainnet' | 'goerli' | 'sepolia';
+
+type InfuraNetworkConfiguration = {
+  network: InfuraNetworkType;
+  infuraProjectId: string;
+  type: NetworkClientType.Infura;
+};
+
+/**
+ * Create a JSON RPC network client for a specific network.
+ *
+ * @param networkConfig - The network configuration.
+ * @returns
```

Comment:
Oh yeah I guess we don't have the full gamut of JSDoc ESLint rules turned on for the extension, sorry. What about:

``` suggestion
 * @returns the provider and block tracker for the network.
```